app/src/main/java/com/example/konrad/decentwork/Registration.java

```diff
@@ -0,0 +1,158 @@
+    ProgressDialog progressDialog;
+
+    private EditText signupInputPassword, signupInputEmail;
+    private Button btnSignUp;
+    private Button btnLinkLogin;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        this.requestWindowFeature(Window.FEATURE_NO_TITLE);
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_registration);
+
+        // Progress dialog
+        progressDialog = new ProgressDialog(this);
+        progressDialog.setCancelable(false);
+        signupInputEmail = (EditText) findViewById(R.id.signup_input_email);
+        signupInputPassword = (EditText) findViewById(R.id.signup_input_password);
+
+
+
+        btnSignUp = (Button) findViewById(R.id.btn_signup);
+        btnLinkLogin = (Button) findViewById(R.id.btn_link_login);
+
+        btnSignUp.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                submitForm();
+            }
+        });
+        btnLinkLogin.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+
+                Intent i = new Intent(getApplicationContext(),Login.class);
+                startActivity(i);
+            }
+        });
+    }
+
+    private void submitForm() {
+//
+//        int selectedId = genderRadioGroup.getCheckedRadioButtonId();
+//        String gender;
+//        if(selectedId == R.id.female_radio_btn)
+//            gender = "Female";
+//        else
+//            gender = "Male";
+
+        registerUser(
+                signupInputEmail.getText().toString(),
+                signupInputPassword.getText().toString());
+    }
+//
+    private void registerUser(final String email, final String password) {
+        // Tag used to cancel the request
+        String cancel_req_tag = "register";
+
+        progressDialog.setMessage("Adding you ...");
+        showDialog();
+
+        StringRequest strReq = new StringRequest(Request.Method.POST,
+                URL_FOR_REGISTRATION, new Response.Listener<String>() {
+
+            @Override
+            public void onResponse(String response) {
+                Log.d(TAG, "Register Response: " + response.toString());
+                hideDialog();
+
+                try {
+                    JSONObject jObj = new JSONObject(response);
+                    boolean error = jObj.getBoolean("error");
+
+                    if (!error) {
+                        String user = jObj.getJSONObject("user").getString("name");
+                        Toast.makeText(getApplicationContext(), "Hi " + user +", You are successfully Added!", Toast.LENGTH_SHORT).show();
+
+                        // Launch login activity
+                        Intent intent = new Intent(
+                                Registration.this,
+                                Login.class);
+                        startActivity(intent);
```

Comment:
Isn't it better if after registration We login user instantly instead of forcing users to login by them selfs?